src/voters/voterFile/voterFile.controller.ts

```diff
@@ -127,4 +132,58 @@ export class VoterFileController {
   ) {
     return this.voterFileFilterService.create(campaign.id, voterFileFilter)
   }
+
+  @Get('filters')
+  @UseCampaign()
+  listVoterFileFilters(@ReqCampaign() campaign: Campaign) {
+    return this.voterFileFilterService.findByCampaignId(campaign.id)
+  }
+
+  @Get('filter/:id')
+  @UseCampaign()
+  async getVoterFileFilter(
+    @Param('id', ParseIntPipe) id: number,
+    @ReqCampaign() campaign: Campaign,
+  ) {
+    const filter: VoterFileFilter | null =
+      await this.voterFileFilterService.findByIdAndCampaignId(id, campaign.id)
+    if (!filter) {
+      throw new NotFoundException('Voter file filter not found')
+    }
+    return filter
+  }
+
+  @Put('filter/:id')
+  @UseCampaign()
+  async updateVoterFileFilter(
+    @Param('id', ParseIntPipe) id: number,
+    @Body() body: UpdateVoterFileFilterSchema,
+    @ReqCampaign() campaign: Campaign,
+  ) {
+    const filter: VoterFileFilter | null =
+      await this.voterFileFilterService.findByIdAndCampaignId(id, campaign.id)
```

Comment:
I don't get it.. 

If you have the ID of the filter from the path parameters of the controller, then why do we have to do a lookup by ID and camping ID? 

ID is a primary key. Is it not?